scheme/sevsnp/evidence_handler.go

```diff
@@ -0,0 +1,530 @@
+		return err
+	}
+
+	if err := validateTA(certChain, provisionedArk); err != nil {
+		return err
+	}
+
+	return validateReportIntegrity(tsm, certChain)
+}
+
+// refvalToComidTriple converts extracted reference values to CoMID value triple
+func refvalToComidTriple(endorsementsStrings []string) (*comid.ValueTriple, error) {
+	var (
+		refValEndorsement *handler.Endorsement
+		rv                comid.ValueTriple
+	)
+
+	for i, e := range endorsementsStrings {
+		var endorsement handler.Endorsement
+
+		if err := json.Unmarshal([]byte(e), &endorsement); err != nil {
+			return nil, fmt.Errorf("could not decode endorsement at index %d: %w", i, err)
+		}
+
+		if endorsement.Type == handler.EndorsementType_REFERENCE_VALUE {
+			refValEndorsement = &endorsement
+			break
+		}
+	}
+
+	if refValEndorsement == nil {
+		return nil, ErrNoProvisionedRV
+	}
+
+	err := json.Unmarshal(refValEndorsement.Attributes, &rv)
+	if err != nil {
+		return nil, err
+	}
+
+	return &rv, nil
+}
+
+// evidenceToComidTriple converts claim set to CoMID value triple
+func evidenceToComidTriple(ec *proto.EvidenceContext) (*comid.ValueTriple, error) {
+	evCorimJson, err := json.Marshal(ec.Evidence.AsMap())
+	if err != nil {
+		return nil, err
+	}
+
+	evComid, err := comidFromJson(evCorimJson)
+	if err != nil {
+		return nil, err
+	}
+
+	return &evComid.Triples.ReferenceValues.Values[0], nil
+}
+
+// compareMeasurements checks if two given comid.Measurement variables are the same.
+func compareMeasurements(refM comid.Measurement, evM comid.Measurement) bool {
+	// RawValue comparison
+	if refM.Val.RawValue != nil {
+		if evM.Val.RawValue == nil {
+			return false
+		}
+
+		refDigest, _ := refM.Val.RawValue.GetBytes()
+		return evM.Val.RawValue.CompareAgainstReference(refDigest, nil)
+	}
+
+	// Digests comparison
+	if refM.Val.Digests != nil {
+		if evM.Val.Digests == nil {
+			return false
+		}
+
+		return evM.Val.Digests.CompareAgainstReference(*refM.Val.Digests)
+	}
+
+	// SVN comparison
+	if refM.Val.SVN != nil {
+		if evM.Val.SVN == nil {
+			log.Debugf("evidence doesn't have SVN")
+			return false
+		}
+
+		if c, ok := evM.Val.SVN.Value.(*comid.TaggedSVN); ok {
+			if r, ok := refM.Val.SVN.Value.(*comid.TaggedSVN); ok {
+				return c.CompareAgainstRefSVN(*r)
+			} else if r, ok := refM.Val.SVN.Value.(*comid.TaggedMinSVN); ok {
+				return c.CompareAgainstRefMinSVN(*r)
+			} else {
+				log.Debugf("unknown refVal SVN type")
+				return false
+			}
+		} else if c, ok := evM.Val.SVN.Value.(*comid.TaggedMinSVN); ok {
+			if r, ok := refM.Val.SVN.Value.(*comid.TaggedMinSVN); ok {
+				return c.Equal(*r)
+			}
+			log.Debugf("can't compare TaggedMinSVN against TaggedSVN")
+			return false
+		} else {
+			log.Debugf("unknown evidence SVN type")
+			return false
+		}
+	}
+
+	return true
+}
+
+func compareTcb(refM comid.Measurement, evM comid.Measurement) bool {
+	if refM.Val.SVN == nil {
+		log.Errorf("reference doesn't have SVN")
```

Comment:
```suggestion
		errors.New("reference doesn't have SVN")
```

Reply:
However, just a high-level comment on the API signature: returning an error rather than just a boolean provides richer semantics to the caller.